src/Parallel/ParallelAnalyser.php

```diff
@@ -198,7 +200,12 @@ public function analyse(
 
 				$job = array_pop($jobs);
 				$process->request(['action' => 'analyse', 'files' => $job]);
-			}, $handleError, function ($exitCode, string $output) use (&$internalErrors, &$internalErrorsCount, $processIdentifier): void {
+			}, $handleError, function ($exitCode, string $output) use (&$someChildEnded, &$peakMemoryUsages, &$internalErrors, &$internalErrorsCount, $processIdentifier): void {
+				if ($someChildEnded === false) {
+					$peakMemoryUsages['main'] = memory_get_usage(true);
```

Comment:
This is the point of time where phpstan consumes most memory (I believe in >99% cases), so I add main thread memory at that specific time.